File: app/src/main/java/com/example/myapplication/activities/SettingActivity.java
package com.example.myapplication.activities;

import android.app.Activity;
import android.app.Application;
import android.content.SharedPreferences;
import android.content.res.Configuration;
import android.os.Bundle;
import android.preference.ListPreference;
import android.preference.Preference;
import android.preference.PreferenceFragment;
import android.preference.PreferenceManager;
import android.support.v7.app.AppCompatActivity;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;

import com.example.myapplication.R;

import java.util.Locale;

public class SettingActivity extends AppCompatActivity {
    @Override
    public void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);

        getFragmentManager()
                .beginTransaction()
                .add(R.id.prefs_content, new SettingsFragment())
                .commit();
    }
    public static class SettingsFragment extends PreferenceFragment {
        @Override
        public void onCreate(Bundle savedInstanceState) {
            super.onCreate(savedInstanceState);

            addPreferencesFromResource(R.xml.settings);

        }

        @Override
        public void onResume() {
            super.onResume();
            SharedPreferences prefs= PreferenceManager.getDefaultSharedPreferences(getActivity());
            ListPreference list = (ListPreference) findPreference("languages");
            ListPreference list2 = (ListPreference) findPreference("colors");
            ListPreference list3 = (ListPreference) findPreference("database");
            Configuration configuration = new Configuration();
            getActivity().setTitle(getString(R.string.action_settings));

            list.setSummary(list.getValue());
            list2.setSummary(list2.getEntry());
            list3.setSummary(list3.getValue());

            list.setOnPreferenceChangeListener((preference, o) -> {
                list.setSummary(o.toString());
                if (o.toString().equals("Русский")){
                    Locale loc = new Locale("ru");
                    configuration.setLocale(loc);
                }
                if (o.toString().equals("English")){
                    Locale loc = new Locale("en");
                    configuration.setLocale(loc);
                }

                getActivity().getBaseContext().getResources().updateConfiguration(configuration,
                        getActivity().getBaseContext().getResources().getDisplayMetrics());
                getActivity().recreate();
                return true;
            });

            list2.setOnPreferenceChangeListener((preference, o) -> {
                list2.setSummary(o.toString());
                list2.setValue(o.toString());
                getActivity().recreate();
                return false;
            });

            list3.setOnPreferenceChangeListener((preference, o) -> {
                list3.setSummary(o.toString());
                return true;
            });
        }
    }
}

File: app/src/main/java/com/example/myapplication/activities/SettingActivity.kt
package com.example.myapplication.activities

import android.content.res.Configuration
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.preference.PreferenceFragmentCompat

import com.example.myapplication.R

import java.util.Locale

class SettingActivity : AppCompatActivity() {
    public override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)

        supportFragmentManager
                .beginTransaction()
                .replace(R.id.prefs_content, SettingsFragment())
                .commit()
    }

    class SettingsFragment : PreferenceFragmentCompat() {
        /**
         * Called during [.onCreate] to supply the preferences for this fragment.
         * Subclasses are expected to call [.setPreferenceScreen] either
         * directly or via helper methods such as [.addPreferencesFromResource].
         *
         * @param savedInstanceState If the fragment is being re-created from
         * a previous saved state, this is the state.
         * @param rootKey If non-null, this preference fragment should be rooted at the
         * [androidx.preference.PreferenceScreen] with this key.
         */
        override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
            setPreferencesFromResource(R.xml.settings, rootKey)

        }

        override fun onCreate(savedInstanceState: Bundle?) {
            super.onCreate(savedInstanceState)

        }

        override fun onResume() {
            super.onResume()
            val list = findPreference("languages")
            val list2 = findPreference("colors")
            val list3 = findPreference("database")
            val configuration = Configuration()
            activity!!.title = getString(R.string.action_settings)

            list.summary = list.sharedPreferences.getString(list.key, "")
            list2.summary = list2.sharedPreferences.getString(list2.key, "")
            list3.summary = list3.sharedPreferences.getString(list3.key, "")

            list.setOnPreferenceChangeListener { _, o ->
                if (o.toString() == "Русский") {
                    val loc = Locale("ru")
                    configuration.setLocale(loc)
                }
                if (o.toString() == "English") {
                    val loc = Locale("en")
                    configuration.setLocale(loc)
                }

                activity!!.baseContext.createConfigurationContext(configuration)
                activity!!.recreate()
                true
            }

            list2.setOnPreferenceChangeListener { _, o ->
                activity!!.recreate()
                true
            }

            list3.setOnPreferenceChangeListener { _, o ->
                activity!!.recreate()
                true
            }
        }
    }
}
